sql/wsrep_high_priority_service.cc

```diff
@@ -168,6 +168,10 @@ Wsrep_high_priority_service::Wsrep_high_priority_service(THD* thd)
    */
   thd->variables.option_bits|= OPTION_BIN_LOG;
 
+  /* Allow applying in a transaction read-only context */
```

Comment:
For wsrep, I would save original values and then set these and restore global default after use.